src/coordinator/generated/proto/rpc/query.proto

```diff
@@ -62,8 +62,36 @@ message FetchResult {
 	repeated Series series = 1;
 }
 
+message Segment {
+	bytes head = 1;
+	bytes tail = 2;
+}
+
+message Segments {
+	Segment merged = 1;
+	repeated Segment unmerged = 2;
+}
+
+message CompressedValuesReplica {
+	repeated Segments segments = 1;
+}
+
+message CompressedTag {
+	bytes name = 1;
+	bytes value = 2;
+}
+
+message CompressedDatapoints {
+	bytes namespace = 1;
+	int64 startTime = 2;
+	int64 endTime = 3;
+	repeated CompressedTag tags = 4;
+	repeated CompressedValuesReplica replicas = 5;
+}
+
 message Series {
-	string name = 1;
+	bytes id = 1;
```

Comment:
so only in some cases tags,  datapoints,  compressed will be set ? Is it possible to change it so that:

message Series {
 CompressedSeries compressed
 DecompressedSeries decompressed
}

once we get the right struct, we can be assured all the underlying fields would be present ? Otherwise, in the current design its a little confusing to me whether things like tags would be present for compressed case

Reply:
Yeah, I'm doubling down on tags briefly; got some further work to actually add this to the write path which would consolidate the duplicate values